src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -697,27 +697,29 @@ public static String center(String str, final int size, String padStr) {
      * @return String without newline, {@code null} if null String input
      */
     public static String chomp(final String str) {
-        if (isEmpty(str)) {
+        final int length = length(str);
+
+        if (length == 0) {
             return str;
         }
 
-        if (str.length() == 1) {
+        if (length == 1) {
             final char ch = str.charAt(0);
             if (ch == CharUtils.CR || ch == CharUtils.LF) {
                 return EMPTY;
             }
             return str;
         }
 
-        int lastIdx = str.length() - 1;
+        int lastIdx = length - 1;
         final char last = str.charAt(lastIdx);
 
-        if (last == CharUtils.LF) {
-            if (str.charAt(lastIdx - 1) == CharUtils.CR) {
-                lastIdx--;
+        if (last != CharUtils.LF) {
+            if (last != CharUtils.CR) {
+                return str;
             }
-        } else if (last != CharUtils.CR) {
-            lastIdx++;
+        } else if (str.charAt(lastIdx - 1) == CharUtils.CR) {
+            --lastIdx;
```

Comment:
Better to use lastIdx-- here - don't make unnecessary changes

Reply:
> Better to use lastIdx-- here - don't make unnecessary changes

@sebbASF OK